### PR TITLE
Ticket_url

### PR DESCRIPTION
**Pull request by AnushaTandon**

### CODE REVIEW

File: app/models/project.rb [hunk 1]
```diff
@@ -7,9 +7,12 @@ class Project < ApplicationRecord
   validate :end_after_start
 
   private
+    
     def end_after_start
-      return if end_date < start_date 
-        errors.add(:end_date, "End date must be after the start date") 
+      return if end_date.blank? || start_date.blank?
```
Comment (ror-mahesh) [inline app/models/project.rb:12]: we already validate presence true.

File: app/views/tasks/_form.html.erb [hunk 2]
```diff
@@ -30,6 +30,11 @@
                                             <%= form.text_field :ticket_id, class: 'form-control' %>
                                     </div>
 
+                                     <div class="field mt-3 mb-3">
+                                        <%= form.label :ticket_url %>
+                                            <%= form.text_field :ticket_url, class: 'form-control' %>
```
Comment (ror-mahesh) [inline app/views/tasks/_form.html.erb:35]: can we also add url validation for ticket_url.

**Review by ror-mahesh — CHANGES_REQUESTED** [review]
@AnushaTandon resolve conflict.

File: app/models/project.rb (hunk 1 above)
Comment (AnushaTandon) [reply app/models/project.rb:12]: sir vo is line ke bina error de raha hai.

File: app/views/tasks/_form.html.erb (hunk 2 above)
Comment (AnushaTandon) [reply app/views/tasks/_form.html.erb:35]: yes, sir..

File: app/models/project.rb (hunk 1 above)
Comment (ror-mahesh) [reply app/models/project.rb:12]: because we didn't add validation of presence  for  start date and end date.

**Review by ror-mahesh — CHANGES_REQUESTED** [review]

File: app/models/task.rb [hunk 3]
```diff
@@ -6,10 +6,11 @@ class Task < ApplicationRecord
...
   has_many :notes, dependent: :destroy
   validate :end_after_start
-  validate :start_date_cannot_be_in_the_past
   validate :ticket_should_be_compulsory
-  validates_length_of :title, :in => 5..32, presence: true
-  validates_length_of :description, :in => 5..20, presence: true
+  validates :title, presence: true
+  validates :ticket_url, presence: true
+  validates :description, presence: true
+  validates :start_date, :end_date, presence: true
```
Comment (ror-mahesh) [inline app/models/task.rb:13]: can we add all this in single line.

**Review by ror-mahesh — CHANGES_REQUESTED** [review]
@AnushaTandon Validation is missing for ticket_url,

**Review by ror-mahesh — APPROVED** [review]

**Review by rorshaan — APPROVED** [review]

**Pull request merged**